email_messaging_frequency.view.lkml

```diff
@@ -25,15 +27,17 @@ view: email_messaging_frequency {
                     (deliveries.canvas_step_api_id)=(clicks.canvas_step_api_id))
         LEFT JOIN DATALAKE_SHARING.CHANGELOGS_CAMPAIGN_SHARED AS campaign ON deliveries.campaign_id = campaign.id
           and deliveries.time >= campaign.time
+        LEFT JOIN DATALAKE_SHARING.CHANGELOGS_CANVAS_SHARED as canvas on deliveries.canvas_id = canvas.id
+          and deliveries.time >= canvas.time
       WHERE
-      {% condition campaign_name %} campaign.campaign_name {% endcondition %}
---      AND
---      {[]% condition canvas_name %} deliveries.canvas_name {[]% endcondition %}
+      {% condition campaign_name %} campaign.name {% endcondition %}
+      AND
+      {% condition canvas_name %} canvas_name {% endcondition %}
       AND
       {% condition message_variation_id %} deliveries.message_variation_api_id {% endcondition %}
 --      AND
---      {[]% condition canvas_name %} deliveries.canvas_step_id {[]% endcondition %}
-      qualify row_number() over (partition by delivered_id ORDER BY campaign.time DESC) = 1)
+--      {% condition canvas_name %} deliveries.canvas_step_id {% endcondition %}
+      qualify row_number() over (partition by delivered_id ORDER BY campaign.time, canvas.time DESC) = 1)
```

Comment:
this probably doesn't matter, but should we order by canvas first then campaign just to follow the hierarchy in braze?